toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/AuthenticatorState.kt

```diff
@@ -164,29 +191,74 @@ private class AuthenticatorStateImpl(
         }
     }
 
+    override fun dismissAll() {
+        pendingOAuthUserSignIn.value?.cancel()
+        pendingUsernamePasswordChallenge.value?.cancel()
+        pendingClientCertificateChallenge.value?.onCancel?.invoke()
+        pendingServerTrustChallenge.value?.distrust()
+        pendingIapSignIn.value?.cancel()
+    }
+
+    override fun updateIapConfigurations(iapConfigurations: List<IapConfiguration>) {
+        _iapConfigurations.value = iapConfigurations
+    }
+
     override suspend fun handleArcGISAuthenticationChallenge(challenge: ArcGISAuthenticationChallenge): ArcGISAuthenticationChallengeResponse {
+        return when (challenge.type) {
+            ArcGISAuthenticationChallengeType.Iap -> handleIapChallenge(challenge.requestUrl)
+            ArcGISAuthenticationChallengeType.OAuthOrToken -> handleOAuthOrTokenChallenge(challenge)
+            else -> handleArcGISTokenChallenge(challenge)
```

Comment:
Best to be explicit here instead of using `else`, so that we get a compile error if a new challenge type would be introduced:
```suggestion
            ArcGISAuthenticationChallengeType.Token -> handleArcGISTokenChallenge(challenge)
```